src/models/MyAccount.ts

```diff
@@ -227,15 +260,25 @@ export default class MyAccount {
 export const MyAccountFactory = async (id: string) => {
   client = await sierraClient()
   const baseQuery = `patrons/${id}`
-  const checkouts = await MyAccount.fetchCheckouts(baseQuery)
-  const holds = await MyAccount.fetchHolds(baseQuery)
-  const patron = await MyAccount.fetchPatron(baseQuery)
-  const fines = await MyAccount.fetchFines(baseQuery)
-  const checkoutBibData = await MyAccount.fetchBibData(
-    checkouts.entries,
-    "item"
-  )
-  const holdBibData = await MyAccount.fetchBibData(holds.entries, "record")
+  let holds, patron, fines, checkoutBibData, checkouts, holdBibData
+  try {
+    checkouts = await MyAccount.fetchCheckouts(baseQuery)
+    holds = await MyAccount.fetchHolds(baseQuery)
+    patron = await MyAccount.fetchPatron(baseQuery)
+    fines = await MyAccount.fetchFines(baseQuery)
+  } catch (e) {
+    throw new MyAccountModelError("fetching patron data", e)
```

Comment:
A Sierra client error wouldn't be thrown on a fetch error. There might be an error thrown by the sierra wrapper node module, but it will be caught in the My Account fetchers and bubble up as a MyAccountError. Sierra client only has one method and only two errors that we're throwing. SieraClientErrors only occur on instantiation.

Reply:
This makes more sense, thanks.